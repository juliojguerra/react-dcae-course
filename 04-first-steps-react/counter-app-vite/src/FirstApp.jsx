import PropTypes from "prop-types";

// const getGreeting = (name) => {
//   return `Hola ${name}`;
// };

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1> {title} </h1>
      {/* <h1>{JSON.stringify(newMessage)}</h1> */}
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  title: "No title",
  subtitle: "No subtitle",
  name: "Julio Guerra",
};
