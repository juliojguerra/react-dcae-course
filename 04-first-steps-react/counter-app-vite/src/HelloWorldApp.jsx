// export function App() {
//   // document.createElement...
//   return <h1> Hello World </h1>;
// }

// export default App;

export const HelloWorldApp = () => {
  return (
    <h1>
      Hello World App
    </h1>
  );
};

// FirstApp
