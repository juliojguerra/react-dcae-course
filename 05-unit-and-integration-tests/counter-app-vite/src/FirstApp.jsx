import React from "react";
import PropTypes from "prop-types";

export function FirstApp({
  title = "Default Title",
  subtitle = "Default Subtitle",
  name = "Default Name",
}) {
  return (
    <div>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </div>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};
