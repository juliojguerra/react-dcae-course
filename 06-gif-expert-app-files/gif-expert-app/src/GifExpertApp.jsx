import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const onAddCategory = (onNewCategory) => {
    console.log(onNewCategory);

    setCategories([...categories, onNewCategory]);
  };

  return (
    <div>
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory
        // setCategories={setCategories}
        onNewCategory={onAddCategory}
      />
      {/* Gifs list */}

      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}

        {/* Gif item */}
      </ol>
    </div>
  );
};
