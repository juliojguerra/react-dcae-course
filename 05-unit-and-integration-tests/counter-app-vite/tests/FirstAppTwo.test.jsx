import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Tests in <FirstApp />", () => {
  const title = "Hi, I'm Goku";
  const subtitle = "I'm a subtitle";

  test("should match with snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show message hi Im goku", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  });

  test("should show title in h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("should show subtitle sent by props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(1);
  });
});
