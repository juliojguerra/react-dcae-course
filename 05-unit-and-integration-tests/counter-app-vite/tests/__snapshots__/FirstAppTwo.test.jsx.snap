// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match with snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hi, I'm Goku
    </h1>
    <p>
      Default Subtitle
    </p>
    <p>
      Default Name
    </p>
  </div>
</div>
`;
